### ISSUE TITLE
.meta files somewhere else

### ISSUE THREAD

**Issue by hboetes** [rakshasa/rtorrent#343]
Is it possible to store the .meta files elsewhere? I think they should be stored in the .session dir, not in the download dir.

**Comment by chros73** [comment]
What meta files are we talking about? The original one (that was put into watch directory) or the one that rtorrent has been created (inside session directory)?
Have you specified a session directory in rtorrent config?

**Comment by hboetes** [comment]
* The `.meta` files, which are created when a magnet url is used.
* Yes I have a session directory configured.

**Comment by omgiafs** [comment]
i just run next script by cron:
```
#!/bin/bash
DIR="ROOT_FOLDER_FOR_ALL_TORRENTS"
find $DIR -name \*.meta -type f -delete
```
This is workaround, but moving *.meta files elsewhere would be better than my method.

**Comment by tobia** [comment]
Can anybody familiar with the source code say if / when it's OK to delete these `.meta` files?
(the ones created in the download dir when a magnet URL is used)

**Comment by hboetes** [comment]
They are basically torrent files, required to download and check the
torrent. So removing them when the torrent is disabled seems fine. This
actually happens when the torrent actually finished. But in case the
torrent never downloaded in the first place an empty file is created and
that file is never removed. IMHO they should be stored in the .session
directory.




# Han
--

**Comment by ArtyumX** [comment]
I also think that .meta should be stored in .session directory.

**Comment by rakshasa** [comment]
This is a change that someone else should easily be able to implement.

Just make a meta torrent dir variable and use session if not set. The time to implement and test is why I'm not doing this, not because I won't accept a patch.

**Comment by chros73** [comment]
A.
> Just make a meta torrent dir variable and use session if not set.

1. It's a good idea, but the ideal would be if with some command parameter we can specify their place per torrent (e.g. a watch directory, that we can have more than one), since the downloaded meta file is tied to the torrent (similar way as a torrent file).

2. Where are these meta files are created in the code? I couldn't find it, neither in rtorrent nor in libtorrent project.

B.

1. There is a bug (probably it was a feature at some point :) ) in the produced meta file: it doesn't include `info` part, it starts with `files` straight away. The result is an invalid file, e.g. `lstor` from `pyrocore` can't process them.

- currently: `d5:filesld6:length`
- correct way: `d4:infod5:filesld6:length`

2. Is it possible to include tracker info (if there's any) in these meta files?
It'd be perfect in this way:

- if there's tracker info then include it and use `.torrent` extension (we get a full valid torrent file)
- if there isn't: then use `.meta` extension (as it is now, with the above fix)

**Comment by rakshasa** [comment]
Meta files are just normal torrents that rtorrent creates, except they have a metadata variable set. (in loading of torrents iirc)

There's some special handling done in libtorrent too iirc, however just start from the point where rtorrent is passed a magnet link to load, and do some grep'ing of the variables it sets.

**Comment by Feriman22** [comment]
Hi,

This issue is still existing and should be store these *.meta files in session folder.

Is there any solution since last commment in this thread?

Thanks!

**Comment by chros73** [comment]
> This issue is still existing and should be store these *.meta files in session folder.

Here's a [workaround](https://github.com/chros73/rtorrent-ps-ch_setup/blob/master/ubuntu-14.04/home/chros73/.rtorrent.rc#L213) that I use in my setup, but helper commands are all over place, so you can't just copy-paste the block.
And this is the [simplified version](https://github.com/rakshasa/rtorrent/issues/212#issuecomment-311134314) of the above.

**Comment by rakshasa** [comment]
Done.

**Comment by hboetes** [comment]
10 years after reporting it.  😱 🥳

**Comment by Feriman22** [comment]
> 10 years after reporting it. 😱 🥳

Yeah, I have been using qbit-nox in Docker for several years, so this is a bit late for me.
